### ISSUE TITLE
String filters and Null values

### ISSUE THREAD

**Issue by kzangeli** [telefonicaid/fiware-orion#2359]
The string filters don't consider attribute/metadata values of type `Null`.
For some reason this case wasn't taken into consideration when implementing the string filters.

**Effort: 1 man day**

**Comment by fgalan** [comment]
To be precisse, this issue refers to the JSON datatype `null`.

**Comment by fgalan** [comment]
Both for `q` and `mq`.

**Comment by kzangeli** [comment]
We might have just forgotten about NULL when implementing string filters.
However, I did a quick test and it failed:

    StringFilter::mongoFilterPopulate()
    {
      ...
      case SfvtNull:
        bb.append("$exists", true).append("$in", BSON_ARRAY("null"));
        break;

That try gives the string "null" in the query: 

    attrs.A1.value.N: { $exists: true, $in: [ "null" ] } }

And trying with NULL:

      case SfvtNull:
        bb.append("$exists", true).append("$in", BSON_ARRAY(NULL));
        break;

Well, that doesn't even compile.
Not very important, leaving it for later.
To fix this, look at servicePath and copy how it's done there:

    servicePath: { $in: [ /^/.*/, null ] }

**Comment by fgalan** [comment]
Not sure, but I remember something like this in a similar case... it may work.

```
    BSONObj empty;
    bb.append("$exists", true).append("$in", BSON_ARRAY(empty));
```

**Comment by fgalan** [comment]
Fixed by PR #2806
